align/compiler/match_graph.py

```diff
@@ -175,7 +175,7 @@ def _group_block_const(self,G1,name):
                     )
                 self._update_sym_const(name, G1, [const.name, *const.instances], inst_name, const_list)
                 self._update_block_const(name, G1, [const.name, *const.instances], inst_name, const_list)
-            #Removing single instances of instances
+            #Removing single instances of instances. TODO from sy: DoNotIdentify might have a single instance!
```

Comment:
again, why check len(instances) > 1?

Reply:
I do not see a check here. Are we talking about the check in _update_block_const?

Reply:
line # 184

Reply:
Will update this:
Need to make it more comprehensive to check for redundant constraints: 
1. constraints requiring more than one instance- order, align, symm
2. constraint requiring atleast one instance do_not_dentify

This is different than schema checks which checks for validity. This is to ensure correct translation of constraints across hierarchy.

Reply:
I think we should clearly define what constraint translation should/not do. and make sure translation is not creating any backdoors to verification (@parijatm ). If a constraint is dropped, there is no way to find out after PnR is done.